Improve course cards

ivanvolti opened the pull request:
Hey Kent! I've implemented the cards, but I'm not quite happy with how they look right now, especially in dark mode (the comparison of the current implementation versus my original design is below). The thing is, to make the cards look good I need to use the new images I added, and because you use Cloudinary I need your help - could you upload the new images there so I can use them in the cards? 

**Current Implementation:**

| Dark Mode | Light Mode |
|-----------|------------|
| ![CleanShot 2025-04-21 at 02 21 42@2x](https://github.com/user-attachments/assets/91f12a03-16bf-4fa2-a73e-7c83bdd6142c) | ![CleanShot 2025-04-21 at 02 21 56@2x](https://github.com/user-attachments/assets/a62822c2-219a-4289-a383-1187b50fbeb6) |

**Original Design:**

| Dark Mode | Light Mode |
|-----------|------------|
| ![after-dark](https://github.com/user-attachments/assets/27720106-d670-4d98-8d03-1babd1b01328) | ![after-light](https://github.com/user-attachments/assets/cba1e3cc-66a2-4d3a-906a-3809ba92c017) |

ivanvolti committed:
Improve course cards

ivanvolti committed:
Add images and gradient

ivanvolti commented:
Hey @kentcdodds! Just wanted to check if you've had a chance to look at this PR. I'm really curious about your thoughts on how it looks and feels overall, especially when you resize the screen.

kentcdodds commented:
I'll give this a look when I get the time. If you would like to provide some screenshots and maybe a screen recording that would speed the process up for me a lot.

Thanks so much for working on this, I think it is really cool.

ivanvolti commented:
@kentcdodds The screenshots are already in the PR description along with some questions for you (think you might've missed them). And here's the screen recording:

https://github.com/user-attachments/assets/1881cab4-14d7-4787-a393-60b15b25c00f

ivanvolti commented:
BTW, would you be open to using container queries via Tailwind? Browser support is excellent now: https://caniuse.com/css-container-queries. This would significantly improve the cards design when rendered outside the fullscreen layout, allowing them to adapt to the actual available horizontal space rather than relying on viewport width.

kentcdodds commented:
Sure!

ivanvolti committed:
Add tailwindcss-container-queries

ivanvolti committed:
Switch course-cards to container query

ivanvolti committed:
Clean up code

ivanvolti commented:
@kentcdodds I switched course-cards to container queries. You can see how it responds to size changes and dark mode in this screen recording: https://share.cleanshot.com/pxtk9nHJ

**Also,** I'm not sure if you read the PR description so I'll ask here just in case: can you check out the images I added and upload them to Cloudinary if you like them, so I can use them for the course cards?

kentcdodds commented:
You'll find them all at: `kentcdodds.com/pages/courses/v2/`

The logo.svg was named `epic-web.svg`.

So:

```
https://res.cloudinary.com/kentcdodds-com/image/upload/v1746462308/kentcdodds.com/pages/courses/v2/epic-web-dark.svg
https://res.cloudinary.com/kentcdodds-com/image/upload/v1746462310/kentcdodds.com/pages/courses/v2/epic-web.svg
https://res.cloudinary.com/kentcdodds-com/image/upload/v1746462314/kentcdodds.com/pages/courses/v2/rocket.png
https://res.cloudinary.com/kentcdodds-com/image/upload/v1746462314/kentcdodds.com/pages/courses/v2/rocket-dark.png
https://res.cloudinary.com/kentcdodds-com/image/upload/v1746462314/kentcdodds.com/pages/courses/v2/trophy-dark.png
https://res.cloudinary.com/kentcdodds-com/image/upload/v1746462315/kentcdodds.com/pages/courses/v2/trophy.png
```

Thanks!

ivanvolti committed:
Switch to new course-cards images

kentcdodds commented on app/utils/theme.tsx:
```diff
@@ -60,7 +60,7 @@ function Themed({
 	const themeToReference = initialOnly ? initialTheme : theme
 
-	return <>{themeToReference === 'light' ? light : dark}</>
+	return <>{themeToReference === 'l' ? light : dark}</>
```
I'm not sure why this was changed. I think this will break things.

ivanvolti replied:
`themeToReference` in Chrome on my Mac returns a character `l` or `d`

kentcdodds commented on app/components/course-card.tsx:
```diff
@@ -1,22 +1,17 @@
-import { motion, useReducedMotion, type Variants } from 'framer-motion'
```
I do still like the motion of the arrows if you can keep that :)

kentcdodds reviewed (changes requested):
Thanks! I'm excited to get this merged soon!

ivanvolti committed:
Update Epic Web card image

ivanvolti committed:
Move image classes to images

ivanvolti committed:
Improve course cards styles

ivanvolti committed:
Recover arrow motion

kentcdodds commented:
This is way cool and I look forward to merging soon!

ivanvolti committed:
Remove @ts-expect-error framer-motion directives

ivanvolti commented:
I updated `framer-motion` for better React 19 support. I've tested several pages and didn't notice any errors, but let me know if you'd prefer to keep the old version (or if you'd rather upgrade in a separate PR).

I've also added the arrow motion animation - you can see it in action in this video demo:
https://github.com/user-attachments/assets/bf393474-c496-4bfa-9587-28c5c9d96787

And here's how the other course cards look now:  
https://github.com/user-attachments/assets/44a06ec0-8b46-4cf7-93b6-e8c5cf37e3e4

kentcdodds reviewed (approved):
This is great. Thank you very much!

kentcdodds commented:
Looks like there are a few things that still need to be handled for typescript

ivanvolti committed:
Get rid of the rest @ts-expect-error

ivanvolti commented:
@kentcdodds This time I cleaned up all of the typescript stuff properly, I promise 🙂 At least `npm run typecheck` (`tsc -b .`) doesn't find any issues.

kentcdodds commented:
Thanks! This is way cool.
